app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -738,12 +738,33 @@ export default Component.extend(
       );
     },
 
+    @bind
+    _handleGalleryToggleClick(event) {
+      if (!event.target.classList.contains("image-grid-toggle")) {
+        return;
+      }
+
+      const imageGrid = '<div data-disable-image-grid="true"></div>';
```

Comment:
This isn't ideal as a flag for disabling the grid (it pollutes the post contents a little bit), but I'm not sure we have any other options.